### PR TITLE
SlideEditor: Drag n Drop Sortable Slide Components

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

File: client/components/Slide/SlideEditor/index.jsx [hunk 1]
```diff
@@ -194,51 +244,69 @@ export default class SlideEditor extends React.Component {
...
+                                            {componentMenu}
+                                            {edit}
+                                        </Segment>
+                                    ) : (
+                                        <React.Fragment key={index}>
+                                            {display}
+                                        </React.Fragment>
+                                    );
+                                })}
+                            </Sortable>
```
Comment (rwaldron) [inline client/components/Slide/SlideEditor/index.jsx]: This is just a bunch of whitespace changes caused by nesting the component render in `<Sortable>`

File: client/components/Slide/SlideEditor/index.jsx [hunk 2]
```diff
@@ -56,13 +61,54 @@ export default class SlideEditor extends React.Component {
     onComponentChange(index, value) {
         const components = this.state.components.slice();
         components[index] = value;
-        this.setState({ components }, this.updatedState);
+        this.setState({ components }, this.updateState);
+    }
+
+    onChangeComponentOrder(...args) {
```
Comment (evmiguel) [inline client/components/Slide/SlideEditor/index.jsx]: Nice, draggable components!

File: client/components/Slide/SlideEditor/index.jsx [hunk 3]
```diff
@@ -56,13 +61,54 @@ export default class SlideEditor extends React.Component {
...
+        const to = components[toIndex];
+        if (from && to) {
+            components[toIndex] = from;
+            components[fromIndex] = to;
+        }
+
+        const currentComponentIndex = toIndex;
+
+        this.setState({ components, currentComponentIndex }, this.updateState);
+        this.props.updateEditorMessage('Component moved!');
```
Comment (evmiguel) [inline client/components/Slide/SlideEditor/index.jsx]: This works well.

File: client/components/Slide/SlideEditor/index.jsx [hunk 4]
```diff
@@ -194,51 +244,69 @@ export default class SlideEditor extends React.Component {
...
+                                            items={{
+                                                save: {
+                                                    onClick: (...args) => {
+                                                        this.updateState(
+                                                            ...args
+                                                        );
+                                                    }
+                                                },
+                                                delete: {
+                                                    onConfirm: () => {
```
Comment (evmiguel) [inline client/components/Slide/SlideEditor/index.jsx]: Can we do curryable here? Like we did in the previous PR?

Comment (rwaldron) [reply client/components/Slide/SlideEditor/index.jsx]: Done

**Pull request merged**